### ISSUE TITLE
Elasticsearch_plugin.cpp(324): error C2593: 'operator +' is ambiguous

### ISSUE THREAD

**Issue by cwyyprog** [bitshares/bitshares-core#1290]
When build the develop branch using visual studio 2015.1, an error was encountered:
```C:\bitshares-core\libraries\plugins\elasticsearch\elasticsearch_plugin.cpp(324): error C2593: 'operator +' is ambiguous
 [C:\bitshares-core\libraries\plugins\elasticsearch\graphene_elasticsearch.vcxproj]
  C:\bitshares-core\libraries\fc\include\fc/variant.hpp(656): note: could be 'fc::variant fc::operator +(const fc::vari
  ant &,const fc::variant &)' [found using argument-dependent lookup]
  C:\bitshares-core\libraries\fc\include\fc/time.hpp(16): note: or       'fc::microseconds fc::operator +(const fc::mic
  roseconds &,const fc::microseconds &)' [found using argument-dependent lookup]
  C:\bitshares-core\libraries\fc\include\fc/time.hpp(110): note: or       'fc::time_point fc::operator +(const fc::time
  _point_sec &,const fc::microseconds &)' [found using argument-dependent lookup]
  C:\bitshares-core\libraries\fc\include\fc/uint128.hpp(66): note: or       'fc::uint128 fc::operator +(const fc::uint1
  28 &,const fc::uint128 &)' [found using argument-dependent lookup]
  C:\bitshares-core\libraries\fc\include\fc/safe.hpp(39): note: or       'fc::safe<int64_t> fc::safe<int64_t>::operator
   +(const fc::safe<int64_t> &,const fc::safe<int64_t> &)' [found using argument-dependent lookup]
  C:\bitshares-core\libraries\plugins\elasticsearch\elasticsearch_plugin.cpp(324): note: while trying to match the argu
  ment list '(std::basic_string<char,std::char_traits<char>,std::allocator<char>>, const fc::unsigned_int)'
Done Building Project "C:\bitshares-core\libraries\plugins\elasticsearch\graphene_elasticsearch.vcxproj" (default targe
ts) -- FAILED.
```
and, the boost version is 1.65.1.
What's more, another error is  caused by using sys/socket.h which is not available n windows.
```
C:\bitshares-core\tests\cli\main.cpp(41): fatal error C1083: Cannot open include file: 'sys/socket.h': No such file or
directory [C:\bitshares-core\tests\cli_test.vcxproj]
Done Building Project "C:\bitshares-core\tests\cli_test.vcxproj" (default targets) -- FAILED.
```

**Comment by oxarbitrage** [comment]
was fixed by @abitmore here: https://github.com/bitshares/bitshares-core/commit/e35d628bc7bb03c38db100dd40dd0a31b277c783

this was added to the last `master` but it was after the release tag. please upgrade to last `master`.

thanks.

**Comment by oxarbitrage** [comment]
actually now the changes are also merged to `develop` in https://github.com/bitshares/bitshares-core/pull/1284

so you can clone the branch again and should work. thanks again.

**Comment by abitmore** [comment]
@cwyyprog created a new issue #1292 for the cli_test issue. Thanks. The original issue is fixed by #1284.
